tests/test_adjoint.py

```diff
@@ -26,10 +27,10 @@ def test_gradcheck() -> None:
     state = random_state(MAX_QUBITS)
 
     def adjoint_expfn(values) -> Array:
-        return adjoint_expectation(state, ops, observable, values)
+        return adjoint_expectation(state, ops, observable, values)[0]
```

Comment:
Wouldn't it be better to return the first element within the definition of adjoint_expectation and expectation so the user (particularly having older code) can directly get the results they'd expect?

Reply:
The rationale was to be consistent with the typing/shaping of outputs. Before this MR, horqrux only ever supported single observables, so only ever returned a scalar.

The MR maintains consistency in that if we have `n` observables, then the output is of shape `(n,)`.

So the question here is about whether we juggling consistency with backwards compatibility. I thought for the time being, it makes more sense to prioritise consistency since there are not so many users of `horqrux` yet.